File: src/com/designpatterns/creational/abstractfactory/ShapeFactoryProvider.java
package com.designpatterns.creational.abstractfactory;

public class ShapeFactoryProvider implements IShapeFactoryProvider {

    @Override
    public IShapeFactory getFactory(String factoryType) {
        if (factoryType == null) {
            return null;
        }
        if (factoryType.equalsIgnoreCase(RegularShapeFactory.TAG)) {
            return new RegularShapeFactory();
        } else if (factoryType.equalsIgnoreCase(RoundedShapeFactory.TAG)) {
            return new RoundedShapeFactory();
        } else {
            throw new IllegalArgumentException("Factory type not found");
        }
    }
}

File: src/com/designpatterns/creational/abstractfactory/kotlin/ShapeFactoryProvider.kt
package com.designpatterns.creational.abstractfactory.kotlin

import java.lang.IllegalArgumentException

class ShapeFactoryProvider : IShapeFactoryProvider {

    override fun getFactory(factoryType: String): IShapeFactory {

        return if (factoryType.equals(RegularShapeFactory.TAG, ignoreCase = true)) {
            RegularShapeFactory()
        } else if (factoryType.equals(RoundedShapeFactory.TAG, ignoreCase = true)) {
            RoundedShapeFactory()
        } else {
            throw IllegalArgumentException("Factory type not found")
        }
    }
}